sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/IRetryPolicy.java

```diff
@@ -25,6 +26,20 @@
     Mono<ShouldRetryResult> shouldRetry(Exception e);
 
 
+    int getRetryCount();
+
+    void incrementRetry();
+
+    void captureStartTimeIfNotSet();
+
+    void updateEndTime();
+
+    Duration getRetryLatency();
+
+    void addStatusAndSubStatusCode(int statusCode, int subStatusCode);
+
+    List<int[]> getStatusAndSubStatusCodes();
```

Comment:
please note that shouldRetry() is async and some other methods you added are not async. I don't see a specific issue here, but just please be cautious when mixing these sync vs async methods.

Reply:
Point noted for future reference.